baremetal/machine_manager_test.go

```diff
@@ -0,0 +1,1433 @@
+
+		result, err := actuator.getHost(context.TODO())
+		if err != nil {
+			t.Error(err)
+		}
+		if (result != nil) != tc.ExpectPresent {
+			t.Error(tc.FailMessage)
+		}
+	}
+}
+
+func TestEnsureAnnotation(t *testing.T) {
+	scheme := runtime.NewScheme()
+	err := capi.AddToScheme(scheme)
+
+	if err != nil {
+		log.Printf("AddToScheme failed: %v", err)
+	}
+
+	testCases := map[string]struct {
+		Machine   capi.Machine
+		Host      bmh.BareMetalHost
+		BMMachine capbm.BareMetalMachine
+	}{
+		"Annotation exists and is correct ": {
+			Machine: capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						HostAnnotation: "myns/myhost",
+					},
+				},
+			},
+			Host: bmh.BareMetalHost{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "myhost",
+					Namespace: "myns",
+				},
+			},
+		},
+		"Annotation exists but is wrong": {
+			Machine: capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						HostAnnotation: "myns/wrongvalue",
+					},
+				},
+			},
+			Host: bmh.BareMetalHost{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "myhost",
+					Namespace: "myns",
+				},
+			},
+		},
+		"Annotations are empty": {
+			Machine: capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{},
+				},
+			},
+			Host: bmh.BareMetalHost{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "myhost",
+					Namespace: "myns",
+				},
+			},
+		},
+		"Annotations are nil": {
+			Machine: capi.Machine{},
+			Host: bmh.BareMetalHost{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "myhost",
+					Namespace: "myns",
+				},
+			},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Logf("## TC-%s ##", name)
+		c := fakeclient.NewFakeClientWithScheme(scheme, &tc.Machine)
+		actuator, err := newMachineManager(c, nil, nil, &tc.Machine, &tc.BMMachine)
+		if err != nil {
+			t.Error(err)
+		}
+
+		err = actuator.ensureAnnotation(context.TODO(), &tc.Host)
+		if err != nil {
+			t.Errorf("unexpected error %v", err)
+		}
+
+		// get the machine and make sure it has the correct annotation
+		machine := capi.Machine{}
+		key := client.ObjectKey{
+			Name:      tc.Machine.Name,
+			Namespace: tc.Machine.Namespace,
+		}
+		err = c.Get(context.TODO(), key, &machine)
+		if err != nil {
+			t.Error(err)
+		}
+
+		annotations := machine.ObjectMeta.GetAnnotations()
+		if annotations == nil {
+			if !(strings.Contains(name, "Annotations are empty") || strings.Contains(name, "Annotations are nil")) {
+				t.Error("no annotations found")
+			}
+		}
+		result, ok := annotations[HostAnnotation]
+		if !ok {
+			if !(strings.Contains(name, "Annotations are empty") || strings.Contains(name, "Annotations are nil")) {
+				t.Error("host annotation not found")
+			}
+		}
+
+		if result != tc.Machine.ObjectMeta.Annotations[HostAnnotation] {
+			t.Errorf("host annotation has value %s, expected \"%s\"",
+				result, tc.Machine.ObjectMeta.Annotations[HostAnnotation])
+		}
+	}
+}
```

Comment:
This is all about Baremetalmachine too, we do not have to give a Machine, it should be nil